src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/hearingsbyhearingtype/HearingsByHearingTypeReportTest.java

```diff
@@ -0,0 +1,297 @@
+    @Before
+    public void setup() {
+
+        DateListedTypeItem dateListedTypeItem = createHearingDateListed("1970-06-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        List<HearingTypeItem> hearings = createHearingCollection(createHearing(HEARING_TYPE_JUDICIAL_HEARING, "JM",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "1", "lead1" ));
+        dateListedTypeItem = createHearingDateListed("2021-07-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        hearings = createHearingCollection(createHearing(HEARING_TYPE_JUDICIAL_REMEDY, "Hybrid",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "2", "lead2"));
+        dateListedTypeItem = createHearingDateListed("2021-06-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        hearings = createHearingCollection(createHearing(HEARING_TYPE_JUDICIAL_COSTS_HEARING, "Stage 1",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "3", "lead3"));
+        dateListedTypeItem = createHearingDateListed("2020-06-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        hearings = createHearingCollection(createHearing(HEARING_TYPE_PERLIMINARY_HEARING, "Video",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "4", "lead4"));
+        dateListedTypeItem = createHearingDateListed("2020-11-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        hearings = createHearingCollection(createHearing(HEARING_TYPE_PERLIMINARY_HEARING_CM, "Full Panel",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "5", "lead5"));
+        dateListedTypeItem = createHearingDateListed("2021-10-01T00:00:00.000",
+                HEARING_STATUS_HEARD);
+        hearings = createHearingCollection(createHearing(HEARING_TYPE_JUDICIAL_RECONSIDERATION, "Yes",
+                dateListedTypeItem));
+        submitEvents.add(createSubmitEvent(hearings, "6", "lead6"));
+    }
+    @Test
+    public void testReportHeaderAreZeroIfNoCasesExist() {
+
+        ListingDetails listingDetails = new ListingDetails();
+        listingDetails.setCaseTypeId(NEWCASTLE_LISTING_CASE_TYPE_ID);
+        ListingData caseData = new ListingData();
+        listingDetails.setCaseData(caseData);
+        List<SubmitEvent> submitEvents = new ArrayList<>();
+        HearingsByHearingTypeReport report = new HearingsByHearingTypeReport();
+        ListingData listingData = report.processHearingsByHearingTypeRequest(listingDetails, submitEvents);
+        verifyReportHeaderIsZero(listingData);
+    }
+
+    private void verifyReportHeaderIsZero(ListingData listingData) {
+        AdhocReportType adhocReportType = listingData.getLocalReportsSummaryHdr();
+        assertEquals(0, adhocReportType == null || Strings.isNullOrEmpty(adhocReportType.getTotal())?0:Integer.parseInt(adhocReportType.getTotal()));
```

Comment:
This doesn't look good. The key thing in unit tests is readability. Surely situations where adhocReportType is null is something quite different from when there is data

Reply:
refactored